pkgs/development/tools/rust/rust-analyzer/generic.nix

```diff
@@ -1,9 +1,10 @@
-{ lib, stdenv, fetchFromGitHub, rustPlatform, darwin
-, useJemalloc ? false
-, doCheck ? true
-
 # Version specific args
-, rev, version, sha256, cargoSha256 }:
+{ rev, version, sha256, cargoSha256 }:
+
+{ lib, stdenv, fetchFromGitHub, rustPlatform, darwin, cmake
+, useMimalloc ? false
+, doCheck ? true
+}:
```

Comment:
Could you also make this a simple function instead of a nested one?

Reply:
Is it necessary? Some are just extracted version related parameters, while others are passed through `callPackages` which can be overrided. It's not expected to directly configure rev, version and hashes like other packages in nixpkgs.

Reply:
It's only a matter of style but I found the original version easier to read than the nested version, although they basically do the same thing.

original:
```nix
rust-analyzer-unwrapped = callPackage ./generic.nix rec {
  rev = ...;
  ...
};
```

nested:
```nix
rust-analyzer-unwrapped = callPackage (import ./generic.nix rec {
  rev = ...;
  ...
}) {};
```